File: doc/codes/python/examples/api/compute/modulation/fourier.py
from autd3.units import Hz
from autd3_modulation import (
    FourierOption,
    SineComponent,
    SineOption,
    fourier,
    modulation_buffer,
)

option = (
    # ANCHOR: option
    FourierOption(
        scale_factor=None,
        clamp=False,
        offset=0x00,
    )
    # ANCHOR_END: option
)
out = modulation_buffer()

# Shown standalone in the SineComponent section of the docs.
# ANCHOR: components
SineComponent(
    100.0 * Hz,
    SineOption(),
)
# ANCHOR_END: components

components = [
    SineComponent(
        100.0 * Hz,
        SineOption(),
    )
]
# ANCHOR: api
fourier(components, option, out)
# ANCHOR_END: api

File: doc/codes/csharp/examples/api/compute/modulation/fourier.cs
using AUTD3;
using static AUTD3.Units;

namespace DocSamples.ApiComputeModulationFourier;

internal static class Sample
{
    internal static void Run()
    {
        float? scaleFactor = null;
        var clamp = false;
        byte offset = 0x00;
        var option =
            // ANCHOR: option
            new FourierOption(
                scaleFactor,
                clamp,
                offset
            )
            // ANCHOR_END: option
            ;
        var dst = Modulation.ModulationBuffer();

        // Shown standalone in the SineComponent section of the docs.
        // ANCHOR: components
        new SineComponent(
            100 * Hz,
            new SineOption()
        );
        // ANCHOR_END: components

        var components = new[]
        {
            new SineComponent(100 * Hz, new SineOption()),
        };
        // ANCHOR: api
        Modulation.Fourier(components, option, dst);
        // ANCHOR_END: api
    }
}
